kore/test/Test/Kore/Step/Substitution.hs

```diff
@@ -130,8 +127,8 @@ test_mergeAndNormalizeSubstitutions =
         $ do
             let expect = Left $ UnificationError $ unsupportedPatterns
                     "Unknown unification case."
-                    (mkElemVar Mock.y)
                     (Mock.constr10 (mkElemVar Mock.y))
+                    (mkElemVar Mock.y)
```

Comment:
Why this change and the similar one at line 222?

Reply:
`simplifyAnds` was sensitive to the order of its arguments, even though that order has no semantic significance. The way I refactored normalization had the effect of changing the argument order. To avoid this problem in the future, I made `simplifyAnds` sort its arguments before unifying; this is the order with the arguments sorted.